scanspec/service.py

```diff
@@ -1,19 +1,114 @@
-from typing import Any
+import base64
+from dataclasses import dataclass
+from typing import Any, List, Optional
 
 import aiohttp_cors
+import graphql
 from aiohttp import web
-from apischema.graphql import graphql_schema
+from apischema.graphql import graphql_schema, resolver
 from graphql_server.aiohttp.graphqlview import GraphQLView, _asyncify
+from numpy import array2string, dtype, float64, frombuffer, ndarray
 
+from scanspec.core import Path
 from scanspec.specs import Spec
 
+# See confluence page for clarity on naming conventions and data structure:
+# https://confluence.diamond.ac.uk/display/SSCC/Data+Structure+and+Naming+Conventions
```

Comment:
Can the outside world see confluence? Is there a description of naming conventions somewhere in the docs here?

Reply:
No, only JIRA is publicly viewable. There's an explanation in the docs about what dimensions are but nothing more. I could look into adding a bit of a glossary if that would be beneficial?

Reply:
I think that would be good. This probably goes in the "Explanations" side of the docs, but with glossary definitions put in the API (by putting them in the docstrings of the classes)